src/Servers/IIS/IIS/src/Core/IISHttpContext.IO.cs

```diff
@@ -122,7 +122,9 @@ private async Task ReadBody()
 
                     if (_consumedBytes > MaxRequestBodySize)
                     {
+#pragma warning disable CS0618 // Type or member is obsolete
                         BadHttpRequestException.Throw(RequestRejectionReason.RequestBodyTooLarge);
```

Comment:
Lets move the static Throw API to a non obsolete internal type so we don't have to use pragma everywhere. I know that's what you were trying to do with the KestrelBadHttpRequestException, but I think we only need to move the static API.